docs/pages/kubernetes-access/discovery/google-cloud.mdx

```diff
@@ -0,0 +1,464 @@
+
+Install Teleport on the host you are using to run the Kubernetes Service and
+Discovery Service:
+
+(!docs/pages/includes/install-linux.mdx!)
+
+### Create a join token 
+
+The Teleport Discovery Service and Kubernetes Service require an authentication
+token in order to to join the cluster. Generate one by running the following
+`tctl` command:
+
+```code
+$ tctl tokens add --type=discovery,kube
+The invite token: (=presets.tokens.first=)
+This token will expire in 60 minutes.
+
+Run this on the new node to join the cluster:
+
+> teleport start \
+   --roles=discovery,kube \
+   --token=(=presets.tokens.first=) \
+   --ca-pin=(=presets.ca_pin=) \
+   --auth-server=192.0.2.255:3025
+
+Please note:
+
+  - This invitation token will expire in 60 minutes
+  - 192.0.2.255:3025 must be reachable from the new node
+```
+
+Copy the token (e.g., `(=presets.tokens.first=)` above) and save the token in
+`/tmp/token` on the machine that will run the Discovery Service and Kubernetes
+Service, for example:
+
+```code
+$ echo (=presets.tokens.first=) | sudo tee /tmp/token
+# (=presets.tokens.first=)
+```
+
+### Configure the Kubernetes Service and Discovery Service
+
+On the host running the Kubernetes Service and Discovery Service, create a
+Teleport configuration file with the following content at `/etc/teleport.yaml`:
+
+```yaml
+version: v2
+teleport:
+  join_params:
+    token_name: "/tmp/token"
+    method: token
+  auth_servers:
+  - "teleport.example.com:3080"
+auth_service:
+  enabled: off
+proxy_service:
+  enabled: off
+ssh_service:
+  enabled: off
+discovery_service:
+  enabled: "yes"
+  gcp:
+    - types: ["gke"]
+      locations: ["*"]
+      project_ids: ["myproject"] # replace with my project ID
+      tags:
+        "*" : "*"
+kubernetes_service:
+  enabled: "yes"
+  resources:
+  - labels:
+      "*": "*"
+```
+
+Edit this configuration for your environment as explained below.
+
+#### `auth_servers` 
+
+Replace `teleport.example.com:3080` with the host and port of your Teleport
+Proxy Service (e.g., `mytenant.teleport.sh:443` for a Teleport Cloud tenant).
+
+#### `discovery_service.gcp`
+
+Each item in `discovery_service.gcp` is a **matcher** for Kubernetes clusters
+running on GKE. The Discovery Service periodically executes a request to the
+Google Cloud API based on each matcher to list GKE clusters. In this case, we
+have declared a single matcher.
+
+Each matcher searches for clusters that match *all* properties of the matcher,
+i.e., that belong to the specified locations and projects and have the
+specified tags. The Discovery Service registers GKE clusters that match *any*
+configured matcher. 
+
+This means that if you declare the following two matchers, the Discovery Service
+will register clusters in project `myproj-dev` running in `us-east1`, as well as
+clusters in project `myproj-prod` running in `us-east2`, but _not_ clusters in
+`myproj-dev` running in `us-east2`:
+
+```yaml
+discovery_service:
+  enabled: "yes"
+  gcp:
```

Comment:
It will be good to extend AWS and Azure guides with the same detail as GCP

Reply:
I can create an issue for this. Are there any particular areas we should concentrate on when expanding the AWS and Azure guides?

Reply:
I'll add to this issue to include separate roles for the Discovery/Kubernetes Service in the AWS/Azure guides per https://github.com/gravitational/teleport/pull/18582#discussion_r1029443620. I think separate roles are appropriate in this case to clarify that these services do different things and make it easier to run the services separately.

Reply:
I agree with you.